File: backend/src/main/java/com/ems/backend/service/DepartmentService.java
package com.ems.backend.service;

import com.ems.backend.dto.DepartmentDto;
import com.ems.backend.dto.Response;

public interface DepartmentService {

    Response createDepartment(DepartmentDto dto);

    Response getAllDepartments();

    Response getDepartmentById(Long id);

    Response updateDepartment(Long id,DepartmentDto departmentDto);

    Response deleteDepartment(Long id);

    Response assignManager(Long deptId,Long managerId);
}

File: frontend/src/services/departmentService.js
import api from "./api";

const API = "http://localhost:8080/api/departments";

export const getDepartments = () => api.get(API);

export const getDepartmentById = (id) => api.get(`${API}/${id}`);

export const createDepartment = (data) => api.post(API,data);

export const updateDepartment = (id,data) => api.put(`${API}/${id}`,data);

export const deleteDepartment = (id) => api.delete(`${API}/${id}`);

export const assignManager = (deptId,managerId) => api.put(`${API}/${deptId}/manager/${managerId}`);
